lib/dams/dams_object.rb

```diff
@@ -391,6 +382,23 @@ def insertFileFields( solr_doc, cid, files )
           end
 	    }
 	  end
+      def fulltext( object, file, cid )
+        mime_type = file.mimeType.first.to_s.gsub(/;.*/,"")
+        if ["application/pdf", "text/html", "text/plain"].include?(mime_type)
+          fid = (cid != nil) ? "_#{cid}_#{file.id}" : "_" + file.id
+          fid = "fulltext#{fid}" if mime_type == "application/pdf"
```

Comment:
I think the pattern is going to be pretty similar with Sufia/CurationConcerns: text/html can be used directly, but PDFs will need to have text extracted.  It'll be saved to a separate file by a background job, but it will look pretty much the same from the indexer's point-of-view.